shm/Alcohol.cpp

```diff
@@ -0,0 +1,49 @@
+#include "Alcohol.h"
+
+#include <algorithm>
+
+Alcohol::Alcohol(std::string name, size_t amount, size_t base_price, Time* time, size_t power)
+    : Cargo(name, amount, base_price, time), power_(power) {}
+
+//override from Cargo
+std::string Cargo::getName() const {
+    return name_;
+}
+
+//override from Cargo
+size_t Cargo::getAmount() const {
+    return amount_;
+}
+
+//override from Cargo
+size_t Cargo::getBasePrice() const {
+    return basePrice_;
+}
+
+//override from Cargo
+size_t Alcohol::getPrice() const {
+    return basePrice_ * (power_ / max_power);
```

Comment:
All this element are integer type. That means that division will also be integer type. For example if you have power less than max power(like 90) division will give you 0 and price will become 0.